main.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    counter = 0
+    while counter < size
+      yield(self[counter])
+      counter += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    output = []
+    my_each { |ele| output << ele if yield(ele) }
+    output
+  end
+
+  def my_all?
+    output = true
+    my_each do |ele|
+      output = false if yield(ele) != true
+    end
+    output
+  end
+
+  def my_any?
+    output = false
+    my_each do |ele|
+      output = true if yield(ele) == true
+    end
+    output
+  end
+
+  def my_none?
+    output = true
+    my_each do |ele|
+      output = false if yield(ele) == true
+    end
+    puts output
+  end
+
+  def my_count(number = nil)
+    count = 0
+    return size if number.nil? && !block_given?
+    if number.nil? && block_given?
+      my_each do |ele|
+        count += 1 if yield(ele, number) == true
+      end
+    end
+    if !number.nil? && block_given?
+      my_each do |ele|
+        count += 1 if yield(ele, number) == true
+      end
+    end
+    count
+  end
+
+  def my_map(&procs)
+    output = []
+    my_each { |ele| output << (procs.nil? ? yield(ele) : procs.call(ele)) }
+    output
+  end
+
+  def my_inject(startervalue)
```

Comment:
#### `REQUIRED`

### :bug: Let's add a case for when a block is given with no argument. Compare with the built-in method. Learn more [here](https://ruby-doc.org/core-2.6.3/Enumerable.html#method-i-inject)

```ruby
irb(main):015:0> [1,2,3,4,5].inject{ |sum, n| sum + n }
=> 15
irb(main):016:0> [1,2,3,4,5].my_inject{ |sum, n| sum + n }
Traceback (most recent call last):
        5: from /../.rbenv/versions/2.6.3/bin/irb:23:in `<main>'
        4: from /../.rbenv/versions/2.6.3/bin/irb:23:in `load'
        3: from /../.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):16
        1: from /../enumerable/main.rb:72:in `my_inject'
ArgumentError (wrong number of arguments (given 0, expected 1)) 
```